### ISSUE TITLE
Integrate MetadataSerDe from Accumulo Recipes into Mango

### ISSUE THREAD

**Issue by cjnolet** [calrissian/mango#165]
This is something that was created for serializing Map<String,String> into a series of bytes and back to a map again. It was implemented as a strategy pattern so different (faster) implementations could always be written at a later time. 

While this has been extremely useful for Accumulo Recipes, there are folks who want to use it on projects like Kafka to serialize metadata for the message keys. I would like it if we could integrate that here so they don't need to pull in Accumulo dependencies to make that possible.

**Comment by eawagner** [comment]
If we do this, I would like to make something not tied to simply metadata (I mean use a different name), but possibly for defining serialization strategies for complex objects in general.  For instance, a List<String>, or Map<Integer, String> to extend the current need for Map<String, String> being used.  To be clear, I don't want to implement SerDe's for all data structures, just come up with an API that facilitates that.  Similar to how the TypeEncoders can be customized.

I guess this would live in the serialize package?

Also can we make sure all work on this is done via a pull request.  This way we can merge it in fully when its ready.

**Comment by cjnolet** [comment]
Yes, new stuff like this should certainly be done in a pull request.
